packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -386,6 +386,72 @@ function useMediaUploadQueue() {
         isTranscoding: state.queue.some((item) => item.state === 'TRANSCODING'),
         isMuting: state.queue.some((item) => item.state === 'MUTING'),
         isTrimming: state.queue.some((item) => item.state === 'TRIMMING'),
+        isResourceProcessing: (resourceId) =>
+          state.queue.some(
+            (item) =>
+              [
+                'PENDING',
+                'UPLOADING',
+                'TRANSCODED',
+                'TRANSCODING',
+                'MUTING',
+                'MUTED',
+                'TRIMMING',
+                'TRIMMED',
+              ].includes(item.state) &&
+              item.resource.originalResourceId === resourceId
+          ),
+        isResourceProcessingById: (resourceId) =>
+          state.queue.some(
+            (item) =>
+              [
+                'PENDING',
+                'UPLOADING',
+                'TRANSCODED',
+                'TRANSCODING',
+                'MUTING',
+                'MUTED',
+                'TRIMMING',
+                'TRIMMED',
+              ].includes(item.state) && item.resource.id === resourceId
+          ),
+        isResourceUploadingById: (resourceId) =>
+          state.queue.some(
+            (item) =>
+              item.state === 'UPLOADING' && item.resource.id === resourceId
+          ),
+        isResourceTranscodingById: (resourceId) =>
+          state.queue.some(
+            (item) =>
+              item.state === 'TRANSCODING' && item.resource.id === resourceId
+          ),
+        isResourceMutingById: (resourceId) =>
+          state.queue.some(
+            (item) => item.state === 'MUTING' && item.resource.id === resourceId
+          ),
+        isResourceTrimmingById: (resourceId) =>
+          state.queue.some(
+            (item) =>
+              item.state === 'TRIMMING' && item.resource.id === resourceId
+          ),
+        isResourceTranscoding: (resourceId) =>
```

Comment:
Nit: could we move those functions together, i.e. move this below/above `isResourceTranscodingById`? Helps with readability